SAFETY.md

```diff
@@ -15,6 +15,10 @@ Quirks specific to Postgres.
 
 The `palloc*` family of functions may throw a Postgres error but will not return `nullptr`.
 
+You should be careful to avoid using data allocated by Postgres after it has been freed. In particular,
+when writing set returning functions, don't directly return referenced data from the arguments without
+cloning the data into memory owned by Rust.
```

Comment:
```suggestion
cloning the data into memory with a sufficient lifetime (either allocated by Rust through a system memory allocator, or via Postgres using `pg_sys::SPI_palloc` (for SPI connections) or PGX's `PgMemoryContexts` interface.
```

Reply:
FYI. There’s actually a pgx bug here. And I’m working on fixing it.  

Once fixed we won’t need this PR at all.

Reply:
Hmm, my idea for this documentation here is that it should be mostly design-level: it should contain whatever principles we need to be mindful of in order to write PGX correctly, which includes the ideas that exist beyond the current implementation.

Reply:
Most of the suggested changes going back-n-forth are informed by bugged behavior.

Reply:
The real name of this document is probably going to wind up something like `PRIMER_ON_POSTGRES_MEMORY_CONTEXTS.md` tbh.

Reply:
I’ve got the beginnings of a PR that will address this.